x-pack/plugin/security/qa/security-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/QueryApiKeyIT.java

```diff
@@ -175,7 +210,123 @@ public void testQueryShouldRespectOwnerIdentityWithApiKeyAuth() throws IOExcepti
             assertThat(
                 apiKeys.stream().map(m -> (String) m.get("name")).collect(Collectors.toUnmodifiableSet()),
                 equalTo(Set.of("power-key-1", "limit-key-1")));
+            apiKeys.forEach(k -> assertThat(k, not(hasKey("_sort"))));
         });
+
+    }
+
+    public void testPagination() throws IOException {
+        final String authHeader = randomFrom(API_KEY_ADMIN_AUTH_HEADER, API_KEY_USER_AUTH_HEADER);
+        final int total = randomIntBetween(8, 12);
+        final List<String> apiKeyIds = new ArrayList<>(total);
+        for (int i = 0; i < total; i++) {
+            apiKeyIds.add(createApiKey(String.format(Locale.ROOT, "k-%02d", i), null, authHeader).v1());
+        }
+
+        final int from = randomIntBetween(0, 3);
+        final int size = randomIntBetween(2, 5);
+        final int remaining = total - from;
+        final String sortField = randomFrom("name", "creation");
```

Comment:
Doens't this assume that no to API keys have the same creation time?
Are we sure that's always going to be true in this test?

Reply:
I added `_doc` as a 2nd sort field. Since keys are created sequential and with `wait_for`, I don't think it is possible to have the same creation time. But it's a good place to exercise `_doc` some more and is more future proof (e.g. keys may be created in parallel).